target/arm/cpu.h

```diff
@@ -218,6 +224,18 @@ typedef struct ARMPACKey {
 } ARMPACKey;
 #endif
 
+#ifdef TARGET_CHERI
+#include "cheri-lazy-capregs-types.h"
+#define AARCH_REG_TYPE aligned_cap_register_t
+#else
+#define AARCH_REG_TYPE uint64_t
+#endif
+
+#ifdef TARGET_CHERI
+#define N_BANK_WITH_RESTRICTED 5
+#else
+#define N_BANK_WITH_RESTRICTED 4
+#endif
 
 typedef struct CPUARMState {
     /* Regs for current mode.  */
```

Comment:
Well, morello does not support 32 bit mode, but sadly this can't be ifdef'd out without a factor / an ugly define hack. There is lots of 32-bit supporting code that will not compile if that array is not there, even though it well never get used in practice.

Reply:
:( It's a shame upstream can't just operate on the 64-bit registers in 32-bit mode (with maybe a special register for PC), then it wouldn't be such an issue

Reply:
A comment there says "32/64 switch only happens when taking and returning from exceptions so the overlap semantics are taken care of then instead of having a complicated union". Apparently whoever wrote that thought that remembering to sync state on every exception / return is somehow less complicated than typing the word union?